Assets/Engine/Interactions/IntentManager.cs

```diff
@@ -19,13 +19,27 @@ public class IntentManager : MonoBehaviour
     public Color colorHarm;
     public Color colorHelp;
 
+    private Button IntentButton;
+
     public enum IntentType
     {
         Help,
         Harm
     }
+
+    void Start()
+    {
+        IntentButton = GetComponent<Button>();
+        IntentButton.onClick.AddListener(TaskOnClick);
```

Comment:
Why not simply add SelectIntent rather than TaskOnClick? That way, you could actually remove the entire TaskOnClick method.

Reply:
That's the way it's done in `ToggleBodyTarget.cs`, so not sure, I just implemented the same logic. 

I don't think it's necessarily bad in case your button can trigger different methods but if you want me to remove it I can do it no problem !

Reply:
No worries. I'm happy enough to leave it as is. At least it's consistent!